### ISSUE TITLE
Check if LDAP works as expected

### ISSUE THREAD

**Issue by mario** [nextcloud/android#1257]
Check if search, favorites (and favoriting process) work as expected when LDAP is the backend.

If not, we probably need to get internal ID from somewhere instead of using the username.

**Comment by mario** [comment]
Ok, so this is a serious issue and affect existing accounts too so we need a migration path.

Two use-cases:
1) open an app with accounts already in

Steps:
- poke /ocs/v2.php/cloud/user for each of the accounts
- replace username in account manager with "id" value found here:

https://usercontent.irccloud-cdn.com/file/8L6mFbyc/Bildschirmfoto%202017-08-02%20um%2009.02.00.png

Make sure if this procedure fails for any account, that it's repeated properly.

2) open an app with no accounts
- once you log-in, if success, poke /ocs/v2.php/cloud/user and store "id" as username
- only then proceed

cc @rullzer for confirmation
cc @tobiasKaminsky for potential implementation, since I have other issues to handle

**Comment by mario** [comment]
Also cc @MorrisJobke since he likes to be involved in everything :P

**Comment by MorrisJobke** [comment]
> Check if search, favorites (and favoriting process) work as expected when LDAP is the backend.

Why is this LDAP related?

You could also simulate this issue on android if you login via our email address. To do so you need to:

* setup an user account
* assign an email to this
* use the email address + password instead of username + password to log in into the Android app
* :boom:


Or am I wrong?

**Comment by mario** [comment]
Sure. But the solution outlined above should solve this.

**Comment by MorrisJobke** [comment]
> Sure. But the solution outlined above should solve this.

Correct. Just wanted to make it easier for you to test this, because then you don't need an LDAP setup.

**Comment by mario** [comment]
You are awesome :)

**Comment by rullzer** [comment]
Yes should work.

Or you do it the dav way and just fetch the current principal:

```xml
<d:propfind xmlns:d="DAV:">
  <d:prop>
     <d:current-user-principal />
  </d:prop>
</d:propfind>
```

**Comment by MorrisJobke** [comment]
See also https://github.com/nextcloud/ios/issues/331#issuecomment-319905716 for a detailed explanation.

**Comment by tobiasKaminsky** [comment]
Will do it on Friday

**Comment by tobiasKaminsky** [comment]
With the new login flow the app gets the correct user id, even when login is done via email.

So this affects only 
- existing accounts
- new accounts on NC<12

**Comment by AndyScherzinger** [comment]
merged for 1.5.0
